### PR TITLE
[Feature][Transform] Support merge shading-tables to one table

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

[Transform] Support merge shading-tables to one table

### How was this patch tested?
Added

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/java/org/apache/seatunnel/e2e/transform/TestTableMergeIT.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.testcontainers.containers.Container;
+
+import java.io.IOException;
+
+public class TestTableMergeIT extends TestSuiteBase {
+
+    @DisabledOnContainer(
+            value = {},
+            type = {EngineType.SPARK},
+            disabledReason = "Currently SPARK do not multi table transform")
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/java/org/apache/seatunnel/e2e/transform/TestTableMergeIT.java]: Please rebase dev. Spark supported now.

Comment (hailin0) [reply seatunnel-e2e/seatunnel-transforms-v2-e2e/seatunnel-transforms-v2-e2e-part-2/src/test/java/org/apache/seatunnel/e2e/transform/TestTableMergeIT.java]: fixed

File: seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/serialization/SeaTunnelRowConverter.java [hunk 2]
```diff
@@ -178,9 +178,10 @@ public SeaTunnelRow unpack(GenericRow engineRow) throws IOException {
         SeaTunnelRowType rowType = (SeaTunnelRowType) dataType;
         RowKind rowKind = RowKind.fromByteValue(engineRow.getByte(0));
         String tableId = engineRow.getString(1);
-        Object[] fields = new Object[rowType.getTotalFields()];
-        for (int i = 0; i < fields.length; i++) {
-            fields[i] = reconvert(engineRow.get(i + 2), rowType.getFieldType(i));
+        Object[] fields = new Object[indexes.length];
+        for (int i = 0; i < indexes.length; i++) {
+            int fieldIndex = indexes[i];
```
Comment (hailin0) [inline seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/serialization/SeaTunnelRowConverter.java:183]: Fix mulit-table transform index mapping

cc @Carl-Zhou-CN

Comment (Carl-Zhou-CN) [reply seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/serialization/SeaTunnelRowConverter.java:183]: You were right. It was my oversight

Comment (Hisoka-X) [reply seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/serialization/SeaTunnelRowConverter.java:183]: How about add a unit test to cover it?

Comment (hailin0) [reply seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/serialization/SeaTunnelRowConverter.java:183]: added

**Review by Hisoka-X — APPROVED** [review]

**Review by Carl-Zhou-CN — APPROVED** [review]
LGTM. if pass ci

**Pull request merged**